ote/src/cljs/ote/ui/form_fields.cljs

```diff
@@ -579,3 +608,110 @@
+       [ui/radio-button {:label    "Tämän palvelun tuottamiseen ei osallistu muita yrityksiä."
+                         :value    "none"
+                         :on-click #(select-type :none)}]
+       [ui/radio-button {:label    "Ilmoita web-osoite, josta yritysten tiedot löytyvät csv-tiedostona"
+                         :value    "csv-url"
+                         :on-click #(select-type :csv-url)}]
+       [ui/radio-button {:label    "Lataa yritysten tiedot csv-tiedostona lomakkeelle"
+                         :value    "csv-file"
+                         :on-click #(select-type :csv-file)}]
+       [ui/radio-button {:label    "Lisää yritysten tiedot lomakkeelle lomakkeen kenttien avulla."
+                         :value    "form"
+                         :on-click #(select-type :form)}]]
+
+      (when-not (nil? data)
+        (case company-source
+          :none [:div.row " "]
+          :csv-url [:div
+                    [:div.row (stylefy/use-style style-base/divider)]
+                    [:div.row
+                     (csv-help-text)
+                     [:div.col-md-6
+                     [field {:name            ::t-service/companies-csv-url
+                             :label           (tr [:field-labels :transport-service-common ::t-service/companies-csv-url])
+                             :hint-text       "https://finap.fi/ote/csv/palveluyritykset.csv"
+                             :full-width?  true
+                             :on-blur         on-url-given
+                             :update!         #(update! {::t-service/companies-csv-url %})
+                             :container-class "col-xs-12 col-sm-6 col-md-6"
+                             :type            :string}
+                      companies-csv-url]]]
+
+                     (let [success (if (= :success (get-in data [:csv-count :status]))
+                                     true
+                                     false)
+                           amount (if (get-in data [:csv-count :count])
+                                    (get-in data [:csv-count :count])
+                                    nil)]
+                       (cond
+                         (and data success) [:div.row {:style {:color "green"}} (tr [:csv :parsing-success] {:count amount})]
+                         (and data (= false success)) [:div.row (stylefy/use-style style-base/required-element)
+                                                       (tr [:csv (get-in data [:csv-count :error])])]
+                         :else [:span]))]
+          :csv-file [:div
+                     [:div.row (stylefy/use-style style-base/divider)]
+                     [:div.row
+                      (csv-help-text)
+                      [:div.row {:style {:padding-top "20px"}}
+                       [field {:name      ::t-service/csv-file
+                               :type      :file
+                               :label     "Tiedosto"
```

Comment:
Translation to label

Reply:
This field is not shown to users. We show our custom button instead and this is only here to contain the file name